spec/services/one_off_redirect_fix_service_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+RSpec.describe OneOffRedirectFixService do
+  it "should find the first non-redirect edition and update all redirects in the chain" do
+    edition_1 = create(:redirect_live_edition, destination: "/redirected-path-1", base_path: "/original-base-path")
+    edition_2 = create(:redirect_live_edition, destination: "/redirected-path-2", base_path: "/redirected-path-1")
+    edition_3 = create(:redirect_live_edition, destination: "/redirected-path-3", base_path: "/redirected-path-2")
+    edition_4 = create(:live_edition, base_path: "/redirected-path-3")
+
+    described_class.update_redirects(edition_1)
+    described_class.update_redirects(edition_2)
+    described_class.update_redirects(edition_3)
+    expect(edition_1.reload.redirects.first[:destination]).to eq edition_4.base_path
+    expect(edition_2.reload.redirects.first[:destination]).to eq edition_4.base_path
+    expect(edition_3.reload.redirects.first[:destination]).to eq edition_4.base_path
+  end
+
+  it "should not update redirects for English edition" do
+    document = create(:document)
```

Comment:
Could we make this more explicit by providing `locale: "en"`? Rather than relying on that being set by default? Otherwise it's not really clear how this test is different to the one above it 🤔  Would be nice to have one that tests the opposite behaviour for non-English locales too. I notice in the first test we're just testing the `update_redirects` method directly, which is fine, but we're only really half-testing the `fix_redirects!` method here.